pythran/pythonic/include/__builtin__/enumerate.hpp

```diff
@@ -39,10 +35,14 @@ namespace pythonic
       };
 
       template <class Iterable>
-      struct enumerate : enumerate_iterator<typename Iterable::iterator> {
+      struct enumerate
+          : private Iterable, /* to hold a reference on the iterable */
+            public enumerate_iterator<
+                typename Iterable::iterator> /* to be compatible with
+                                                __builtin__.next*/
+            {
         using iterator = enumerate_iterator<typename Iterable::iterator>;
-        // we need to keep one ref over the enumerated sequence alive
-        Iterable seq;
+        using iterator::operator*;
```

Comment:
On Wed, Dec 30, 2015 at 12:35:31PM -0800, pbrunet wrote:
> >          using iterator = enumerate_iterator<typename Iterable::iterator>;
> > -        // we need to keep one ref over the enumerated sequence alive
> > -        Iterable seq;
> > +        using iterator::operator*;
> 
> Why do you need this operator?

It's implicitly inherited from Iterable *and* enumerate_iterator. This
is the disambiguation.


We inherit from Iterable to store the original sequence and from
enumerate_iterator to be compatible with next(...)